src/acl_kernel.cpp

```diff
@@ -3042,22 +3054,26 @@ int acl_submit_kernel_device_op(cl_event event) {
   acl_forget_proposed_device_ops(doq);
 
   bool need_reprogram = true;
-  if (device->last_bin) {
-    // compare hash of last program that went through device op queue and the
-    // program required by kernel
-    need_reprogram =
-        device->last_bin->get_devdef().autodiscovery_def.binary_rand_hash !=
-        dev_bin->get_devdef().autodiscovery_def.binary_rand_hash;
-  } else {
-    // compare hash of program that is on the device and the program required by
-    // kernel
-    need_reprogram = device->def.autodiscovery_def.binary_rand_hash !=
-                     dev_bin->get_devdef().autodiscovery_def.binary_rand_hash;
-  }
-
-  if (event->context->split_kernel) {
-    // Always reprogram in split kernel mode. This is a temporary workaround.
```

Comment:
Hmmm I might keep this part as is because it is a workaround, it would be easier to remove this when the proper fix is implemented (although I'm not sure what is the issue this code tries to solve).